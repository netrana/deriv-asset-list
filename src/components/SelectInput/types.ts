import { SizeType } from 'antd/lib/config-provider/SizeContext';
import React from 'react';

export type SelectOption = {
  name: string;
  value: number | string;
  className?: string;
};
export interface Props {
  options: SelectOption[];
  title?: string;
  placeholder?: string;
  className?: string;
  bordered?: boolean;
  onChange?: (value: string) => void;
  defaultValue?: string;
  defaultActiveFirstOption?: boolean;
  filterOption?: boolean;
  showSearch?: boolean;
  onSearch?: (value: string) => void;
  notFoundContent?: React.ReactNode;
  onSelect?: (value: string) => void;
  value?: string;
  allowClear?: boolean;
  disabled?: boolean;
  showArrow?: boolean;
  size?: SizeType;
}
