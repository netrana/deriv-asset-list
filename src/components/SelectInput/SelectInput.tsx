import React, { FC } from 'react';
import { Select, Text, SelectOption } from 'components/misc/helpers';
import classNames from 'classnames';

import { Props } from './types';
import styles from './selectInput.module.scss';

const SelectInput: FC<Props> = (props) => {
  const {
    title,
    placeholder,
    options,
    onChange,
    className = '',
    defaultValue,
    bordered,
    allowClear = true,
    onSearch,
    showSearch,
    notFoundContent,
    filterOption,
    onSelect,
    value,
    disabled,
    size,
    showArrow,
  } = props;

  return (
    <div
      className={classNames({
        [styles.select]: true,
        [className]: Boolean(classNames),
      })}
    >
      {!!title ? (
        <div className={styles.title}>
          <Text strong>{title}</Text>
        </div>
      ) : null}
      <Select
        className={className}
        size={size}
        disabled={disabled}
        defaultValue={defaultValue}
        placeholder={placeholder}
        onChange={onChange}
        bordered={bordered}
        allowClear={allowClear}
        onSearch={onSearch}
        showSearch={showSearch}
        notFoundContent={notFoundContent}
        filterOption={filterOption}
        onSelect={onSelect}
        value={value}
        showArrow={showArrow}
      >
        {options.map(({ name, value }, index) => (
          <SelectOption value={value} key={index}>
            {name}
          </SelectOption>
        ))}
      </Select>
    </div>
  );
};

export default SelectInput;
