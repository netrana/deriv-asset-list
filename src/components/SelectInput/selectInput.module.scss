.select{
    :global(.ant-select) {
        width: 100%;
    }
    .title{
        padding-bottom: 8px;
    }
}