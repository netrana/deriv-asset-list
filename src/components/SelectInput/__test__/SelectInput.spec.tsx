import React from 'react';
import render from 'react-test-renderer';

import SelectInput from '../SelectInput';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  FC: jest.fn(),
}));

jest.mock('components/misc/helpers', () => ({
  ...jest.requireActual('components/misc/helpers'),
  Select: 'Select',
  Text: 'Text',
  SelectOption: 'SelectOption',
}));

describe('SelectInput component test', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const tree = render
        .create(
          <SelectInput
            options={[
              {
                name: 'test',
                value: 'test',
              },
            ]}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
