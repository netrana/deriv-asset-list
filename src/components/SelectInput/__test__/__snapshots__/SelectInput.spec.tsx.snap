// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput component test snapshot should match snapshot 1`] = `
<div
  className="select "
>
  <Select
    allowClear={true}
    className=""
  >
    <SelectOption
      value="test"
    >
      test
    </SelectOption>
  </Select>
</div>
`;
